midend/flattenInterfaceStructs.cpp

```diff
@@ -35,18 +44,34 @@ void StructTypeReplacement::flatten(const P4::TypeMap* typeMap,
 }
 
 StructTypeReplacement::StructTypeReplacement(
-    const P4::TypeMap* typeMap, const IR::Type_Struct* type) {
+    const P4::TypeMap* typeMap, const IR::Type* type) {
     auto vec = new IR::IndexedVector<IR::StructField>();
     flatten(typeMap, "", type, vec);
-    replacementType = new IR::Type_Struct(type->name, IR::Annotations::empty, *vec);
+    if (type->is<IR::Type_Struct>()) {
+        replacementType = new IR::Type_Struct(type->to<IR::Type_Struct>()->name,
+                                              IR::Annotations::empty, *vec);
+    } else if (type->is<IR::Type_Header>()) {
+        replacementType = new IR::Type_Header(type->to<IR::Type_Header>()->name,
+                                              IR::Annotations::empty, *vec);
+    }
 }
 
 const IR::StructInitializerExpression* StructTypeReplacement::explode(
     const IR::Expression *root, cstring prefix) {
     auto vec = new IR::IndexedVector<IR::NamedExpression>();
     auto fieldType = ::get(structFieldMap, prefix);
     CHECK_NULL(fieldType);
-    for (auto f : fieldType->fields) {
+    const IR::Type_Struct *st;
+    const IR::Type_Header *ht;
+    bool is_struct = false;
+    if (fieldType->is<IR::Type_Struct>()) {
```

Comment:
you must change your indentation everywhere to 4 spaces.

Reply:
Will do.